src/pages/BlogPostsById.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import posts from '../data/posts.json';
+import {useParams} from "react-router-dom";
+
+
+function BlogPostsById() {
+    const {id} = useParams();
+    const post = posts.filter(post => post.id === id);
```

Comment:
Hoewel filter prima werkt, had je hier beter de find-methode kunnen gebruiken. Omdat je weet dat je op een uniek ID zoekt, hoeft de functie niet verder te zoekken wanneer hij die ene ID gevonden heeft. Bij filter gaat hij de hele array nog af, wat extra rekenkracht/laadtijd kost. In dit geval maakt dat niets uit, maar bij grotere databases kan dit onnodig tijd kosten.